raiden/transfer/channel.py

```diff
@@ -1383,6 +1383,11 @@ def create_sendlockedtransfer(
 
     token = channel_state.token_address
     recipient = channel_state.partner_state.address
+    recipient_metadata = None
+    for route_state in route_states:
+        recipient_metadata = route_state.address_metadata.get(recipient, None)
```

Comment:
Nitpick:
I think it's a little bit weird that in order to send the locked transfer to a specific recipient user-id, you 
**have** to also pass this information to the next node.
I think it would be better to be able to set the attribute after initialisation of the `Send...` classes - so you would e.g. set the recipient_metadata in the `send_lockedtransfer` one layer above.
But allowing this would be more complicated with the frozen data classes.

Reply:
I cannot follow. Could you elaborate on this?

Reply:
never mind, discussed this with Ulo and it is not a good idea